edgelet/iotedged/src/lib.rs

```diff
@@ -673,7 +673,18 @@ fn vol_mount_uri(config: &mut DockerConfig, uris: &[&Url]) -> Result<(), Error>
     // if the url is a domain socket URL then vol mount it into the container
     for uri in uris {
         if uri.scheme() == UNIX_SCHEME {
-            binds.push(format!("{}:{}", uri.path(), uri.path()));
+            let path = uri.to_uds_file_path()
+                .map_err(Error::from)?;
```

Comment:
`?` already does `Into::into` so you shouldn't need the `map_err`

Reply:
I know that now... ;) I'll restructure this...